custom_components/econnect_metronet/config_flow.py

```diff
@@ -116,42 +117,32 @@ async def async_step_init(self, user_input=None):
         """Manage the options."""
         errors = {}
         if user_input is not None:
-            try:
-                parse_areas_config(user_input.get(CONF_AREAS_ARM_HOME), raises=True)
-                parse_areas_config(user_input.get(CONF_AREAS_ARM_NIGHT), raises=True)
-                parse_areas_config(user_input.get(CONF_AREAS_ARM_VACATION), raises=True)
-            except InvalidAreas:
-                errors["base"] = "invalid_areas"
-            except Exception as err:  # pylint: disable=broad-except
-                _LOGGER.error("Unexpected exception %s", err)
-                errors["base"] = "unknown"
-            else:
-                return self.async_create_entry(title="e-Connect/Metronet Alarm", data=user_input)
+            return self.async_create_entry(title="e-Connect/Metronet Alarm", data=user_input)
 
         # Populate with latest changes or previous settings
         user_input = user_input or {}
-        suggest_arm_home = user_input.get(CONF_AREAS_ARM_HOME) or self.config_entry.options.get(CONF_AREAS_ARM_HOME)
-        suggest_arm_night = user_input.get(CONF_AREAS_ARM_NIGHT) or self.config_entry.options.get(CONF_AREAS_ARM_NIGHT)
-        suggest_arm_vacation = user_input.get(CONF_AREAS_ARM_VACATION) or self.config_entry.options.get(
-            CONF_AREAS_ARM_VACATION
-        )
         suggest_scan_interval = user_input.get(CONF_SCAN_INTERVAL) or self.config_entry.options.get(CONF_SCAN_INTERVAL)
+
+        # Generate sectors list for user config options
+        device = self.hass.data[DOMAIN][self.config_entry.entry_id][KEY_DEVICE]
+        sectors = [(item["element"], item["name"]) for _, item in device.items(q.SECTORS)]
+
         return self.async_show_form(
             step_id="init",
             data_schema=vol.Schema(
                 {
                     vol.Optional(
                         CONF_AREAS_ARM_HOME,
-                        description={"suggested_value": suggest_arm_home},
-                    ): str,
+                        default=self.config_entry.options.get(CONF_AREAS_ARM_HOME, []),
```

Comment:
we need to add a default of `[]` otherwise it fails if you don't select at least one option in all the fields.